lib/array.rb

```diff
@@ -0,0 +1,15 @@
+class Array
+  def my_inject starting = nil, symbol = nil
+    starting.nil? ? result = self[0] : result = starting
+    if starting.is_a? Symbol
+      result = self[0]
+      each { |num| result = result.send starting, num }
+    elsif symbol.is_a? Symbol
+      each { |num| result = result.send symbol, num }
```

Comment:
following this method everything breaks and so far not able to fix it :/
So I  created a new method for each statement and then I did 
`return method_with_starting if starting.is_a? Symbol`
three times, one for each method

By the way, I did't know about drop.